tests/asdf_tests/test_core.py

```diff
@@ -0,0 +1,362 @@
+        coords = Q_(np.asarray([2.0, 5.0, 1.0]), "mm")
+    else:
+        coords = Q_(
+            np.asarray(
+                [[2.0, 5.0, 1.0], [1.0, -4.0, 1.2], [0.3, 4.4, 4.2], [1.1, 2.3, 0.2]]
+            ),
+            "mm",
+        )
+
+    if not time_dep_orientation:
+        orientation = tf.rotation_matrix_z(np.pi / 3)
+    else:
+        orientation = tf.rotation_matrix_z(np.pi / 2 * np.array([1, 2, 3, 4]))
+
+    if not time_dep_orientation and not time_dep_coordinates:
+        return tf.LocalCoordinateSystem(orientation=orientation, coordinates=coords)
+
+    time = pd.DatetimeIndex(["2000-01-01", "2000-01-02", "2000-01-03", "2000-01-04"])
+    return tf.LocalCoordinateSystem(
+        orientation=orientation, coordinates=coords, time=time
+    )
+
+
+def are_local_coordinate_systems_equal(
+    lcs_0: tf.LocalCoordinateSystem, lcs_1: tf.LocalCoordinateSystem
+):
+    """
+    Check if 2 local coordinate systems are identical
+
+    Parameters
+    ----------
+    lcs_0 :
+        First local coordinate system
+    lcs_1 :
+        Second local coordinate system
+
+    Returns
+    -------
+    bool:
+        True if both systems are identical, False otherwise
+    """
+    return lcs_0.orientation.identical(
+        lcs_1.orientation
+    ) and lcs_0.coordinates.identical(lcs_1.coordinates)
+
+
+def test_local_coordinate_system_save():
+    """Test if a LocalCoordinateSystem can be written to an asdf file."""
+    lcs_static = get_local_coordinate_system(False, False)
+    lcs_tdp_c = get_local_coordinate_system(False, True)
+    lcs_tdp_o = get_local_coordinate_system(True, False)
+    lcs_tdp_oc = get_local_coordinate_system(True, True)
+    tree = {
+        "lcs_static": lcs_static,
+        "lcs_tdp_c": lcs_tdp_c,
+        "lcs_tdp_o": lcs_tdp_o,
+        "lcs_tdp_oc": lcs_tdp_oc,
+    }
+    with asdf.AsdfFile(
+        tree, extensions=[WeldxExtension(), WeldxAsdfExtension()], copy_arrays=True,
+    ) as f:
+        f.write_to(buffer_lcs)
+        buffer_lcs.seek(0)
+
+
+def test_local_coordinate_system_load():
+    """Test if an xarray.DataArray can be restored from an asdf file."""
+    f = asdf.open(buffer_lcs, extensions=[WeldxExtension(), WeldxAsdfExtension()],)
+    lcs_static_file = f.tree["lcs_static"]
+    lcs_tdp_c_file = f.tree["lcs_tdp_c"]
+    lcs_tdp_o_file = f.tree["lcs_tdp_o"]
+    lcs_tdp_oc_file = f.tree["lcs_tdp_oc"]
+
+    lcs_static_exp = get_local_coordinate_system(False, False)
+    lcs_tdp_c_exp = get_local_coordinate_system(False, True)
+    lcs_tdp_o_exp = get_local_coordinate_system(True, False)
+    lcs_tdp_oc_exp = get_local_coordinate_system(True, True)
+
+    assert are_local_coordinate_systems_equal(lcs_static_file, lcs_static_exp)
+    assert are_local_coordinate_systems_equal(lcs_tdp_c_file, lcs_tdp_c_exp)
+    assert are_local_coordinate_systems_equal(lcs_tdp_o_file, lcs_tdp_o_exp)
+    assert are_local_coordinate_systems_equal(lcs_tdp_oc_file, lcs_tdp_oc_exp)
```

Comment:
Maybe a good place to use [`pytest.parametrize`](https://docs.pytest.org/en/stable/parametrize.html) ?